process: update PR template to include release notes (#13188)
apache/druid: .github/pull_request_template.md

<!-- Some of the aspects mentioned above may be omitted for simple and small changes. -->

#### Release note
<!-- Give your best effort to summarize your changes in a couple of sentences aimed toward Druid users. 

If your change doesn't have end user impact, you can skip this section.

-->

> For tips about how to write a good release note, see [Release notes](../CONTRIBUTING.md#release-notes).

Reviewer: the other links in the template are full links, I'm not certain if relative links work here or not, so to be safe we might want to make it match the others
```suggestion
For tips about how to write a good release note, see [Release notes](https://github.com/apache/druid/blob/master/CONTRIBUTING.md#release-notes).
```

<!-- Check the items by putting "x" in the brackets for the done things. Not all of these items apply to every PR. Remove the items which are not done or not relevant to the PR. None of the items from the checklist below are strictly necessary, but it would be very helpful if you at least self-review the PR. -->

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.

> - [ ] a release note entry added to the PR description.

Reviewer 2: ```suggestion
- [ ] a release note entry in the PR description.
```
nit
